flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartition.java

```diff
@@ -37,11 +40,16 @@
 	/** The parent partition this subpartition belongs to. */
 	protected final ResultPartition parent;
 
+	/** Used to compress buffer to reduce IO. */
+	@Nullable
+	protected final BufferCompressor bufferCompressor;
+
 	// - Statistics ----------------------------------------------------------
 
 	public ResultSubpartition(int index, ResultPartition parent) {
 		this.index = index;
 		this.parent = parent;
+		this.bufferCompressor = parent.getBufferCompressor();
```

Comment:
Do not need maintain `BufferCompressor` variable inside the `ResultSubpartition`, because it can be got directly from `parent` for specific sub partitions.